src/lib.rs

```diff
@@ -275,6 +286,9 @@ where
                 formatter.write_str("---")
             }
             Code(ref text) => {
+                if let Some(shortcut_text) = state.current_shortcut_text.as_mut() {
+                    shortcut_text.push_str(&format!("`{}`", text));
+                }
                 if let Some(text_for_header) = state.text_for_header.as_mut() {
```

Comment:
Right now it's seemingly possible to use the same text twice, once for shortcuts and once for the header. I have a feeling this can't really be possible, so ideally that's represented in the type-system, or in the way the code behaves, at the least there should be an assertion to express this constraint.